### ISSUE TITLE
Questions about soft assignment

### ISSUE THREAD

**Issue by Scott0526** [chenjie20/CVCL#1]
Dear author,

I‘m wondering that  if the initial clustering soft assignment probabilities **h** produced by MLP are mostly incorrect, wouldn't the confident assignment probabilities **p** become even more inaccurate after confidence enhancement? In such a scenario, how does the cluster-level contrastive loss ensure that the allocation process moves towards the correct direction?

Thank you!

**Comment by chenjie20** [comment]
The output features produced by the proposed CVCL model become more similar in a single view during training if they belong to the same cluster. This benefits from the advantages of the deep neural network (DNN) model on data representation learning. Moreover, the output features of a single sample among the multiple views are forced to be more similar because of the cross-view contrastive loss item. In particular, the output features are considered to be the cluster assignments in CVCL. Please don’t hesitate to contact me if you have any further questions. Thank you.

**Comment by Scott0526** [comment]
Thank you very much for your apply!
Moreover, I found that the proposed cluter-level contrastive strategy performed poorly on long-tail datasets such as Caltech101-20(about 0.3 acc). At first, I thought it might be due to the La loss, but even after removing this term, the performance remained not so good. What do you think could be the possible reasons for this issue?

**Comment by chenjie20** [comment]
I have carefully checked this dataset. It contains more than 2300 samples belonging to 20 categories. The recommended combinations of the parameters in CVCL code are not proper for this dataset. I simply tested one group of the parameters for this dataset. More than 50% (ACC) can be obtained by CVCL. Please feel free to contact me if you are interested it. Here is my email: chenjie20@gmail.com  Thank you.

**Comment by chenjie20** [comment]
The following results can be reproduced using the trained model.
ACC=0.637553; NMI=0.599883 PUR=0.665823 ARI=0.633535
Here is one group of the parameters in my source code: 
args.learning_rate = 0.005
args.batch_size = 30
args.seed = 10
args.con_epochs = 100
args.normalized = True

dim_high_feature = 32
dim_low_feature = 16
dims = [16, 32]
lmd = 0.05
beta = 0.1

The related code and the trained model are available at: 
https://www.jiechen.site/downloads/CVCL-Caltech101-20.zip
